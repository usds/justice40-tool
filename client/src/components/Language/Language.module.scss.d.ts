declare namespace LanguageNamespace {
    export interface ILanguageScss {
        languageContainer: string;
        languageContainerMobile: string;
        languageIcon: string;
        languageLink: string;
    }
  }

declare const LanguageScssModule: LanguageNamespace.ILanguageScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: LanguageNamespace.ILanguageScss;
  };

  export = LanguageScssModule;
