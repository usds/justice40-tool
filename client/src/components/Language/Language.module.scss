@use '../../styles/design-system.scss' as *;

@mixin baseLanguageStyles {
  @include u-margin-right(4);
  @include u-display("flex");

  .languageIcon {
    align-self: baseline;
  }

  .languageLink {
    @include u-display("inline-block");
    @include u-margin-left(1.5);
    
    cursor: pointer;
    font-size: .8rem; // government banner text size not a token
    @include u-padding-top('2px');
    
  }
}

.languageContainer {
    @include baseLanguageStyles();
    @include at-media-max("desktop") {
      display: none;
    }
  }
  
  .languageContainerMobile {
    @include baseLanguageStyles();
    @include u-padding-top(3);
    @include at-media("desktop") {
      display: none;
    }
  }
  