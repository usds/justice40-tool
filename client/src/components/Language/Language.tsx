import React from 'react';
import {IntlContextConsumer, changeLocale} from 'gatsby-plugin-intl';

// @ts-ignore
import languageIcon from '/node_modules/uswds/dist/img/usa-icons/language.svg';
import * as styles from './Language.module.scss';

const languageName = {
  en: 'English',
  es: 'EspaÃ±ol',
};

interface ILanguageProps {
  isDesktop: boolean
}

/**
 * Language component that will allow the user to change languages
 *
 * @param {boolean} isDesktop
 * @return {JSX.Element | null}
 */
const Language = ({isDesktop}:ILanguageProps) => {
  return (
    <div className={isDesktop ? styles.languageContainer : styles.languageContainerMobile}>
      <img className={styles.languageIcon} src={languageIcon} alt={'language icon for selecting language'}/>
      <IntlContextConsumer>
        {({languages, language: currentLocale}) =>
          languages.map((language: React.Key | null | undefined) => (
            <a
              href="#"
              className={styles.languageLink ? `usa-link ${styles.languageLink}` : `usa-link`}
              key={language}
              onClick={() => changeLocale(language)}
            >
              {languageName[language]}
            </a>
          ))
        }
      </IntlContextConsumer>
    </div>
  );
};

export default Language;
