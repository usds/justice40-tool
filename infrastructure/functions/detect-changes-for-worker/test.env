REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1