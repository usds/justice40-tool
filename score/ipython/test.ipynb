{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
