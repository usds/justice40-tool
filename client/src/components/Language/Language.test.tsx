import * as React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import Language from './Language';

describe('rendering of the Language component on mobile', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <Language isDesktop={false}/>
      </LocalizedComponent>,
  );

  it('checks if component renders', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('rendering of the Language component on desktop', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <Language isDesktop={true}/>
      </LocalizedComponent>,
  );

  it('checks if component renders', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
