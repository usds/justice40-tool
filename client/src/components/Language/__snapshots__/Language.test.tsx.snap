// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the Language component on desktop checks if component renders 1`] = `<DocumentFragment />`;

exports[`rendering of the Language component on mobile checks if component renders 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="language icon for selecting language"
      src="test-file-stub"
    />
    <a
      class="usa-link"
      href="#"
    >
      English
    </a>
    <a
      class="usa-link"
      href="#"
    >
      Espa√±ol
    </a>
  </div>
</DocumentFragment>
`;
